runtime/lua/vim/lsp/completion.lua

```diff
@@ -170,6 +170,13 @@ local function get_completion_word(item, prefix, match)
       --
       -- Typing `i` would remove the candidate because newText starts with `t`.
       local text = parse_snippet(item.insertText or item.textEdit.newText)
+      if
+        vim.o.completeopt:find('popup')
+        and #(item.documentation or '') == 0
+        and vim.bo.filetype ~= ''
+      then
+        item.documentation = ('```%s\n%s\n```'):format(vim.bo.filetype, text)
```

Comment:
Does `vim.bo.filetype` need to be translated to a treesitter parser name, or does it go by vim filetype name? @clason

Reply:
I don't think treesitter is relevant here? But I might be missing context.

Reply:
he mean use  something like this `vim.treesitter.language.get_lang(vim.bo.filetype)` to get language as markdown code language identifier

Reply:
Then that shouldn't be needed since the injection language lookup has a built-in fallback to filetypes.

Reply:
so just use filetype as well?

Reply:
Yes.

Reply:
(of course, the filetype still needs to be registered for a parser if the names doesn't match, but that's also the case for the explicit lookup here)